packages/app-builder/src/routes/ressources+/settings+/data-model+/tables+/$tableId+/exported-fields.ts

```diff
@@ -0,0 +1,150 @@
+import { isAdmin } from '@app-builder/models/user';
+import { initServerServices } from '@app-builder/services/init.server';
+import { getRoute } from '@app-builder/utils/routes';
+import { type ActionFunctionArgs, redirect } from '@remix-run/node';
+import z from 'zod';
+
+const ingestedDataFieldSchema = z.object({
+  path: z.array(z.string()),
+  name: z.string(),
+});
+
+const exportedFieldSchema = z.union([
+  z.object({ triggerObjectField: z.string() }),
+  z.object({ ingestedDataField: ingestedDataFieldSchema }),
+]);
+
+export async function action({ request, params }: ActionFunctionArgs) {
+  const { authService } = initServerServices(request);
+  const { user, dataModelRepository } = await authService.isAuthenticated(request, {
+    failureRedirect: getRoute('/sign-in'),
+  });
+
+  if (!isAdmin(user)) {
+    return redirect(getRoute('/'));
+  }
+
+  const { tableId } = params;
+  if (!tableId) {
+    return Response.json({ error: 'Table ID is required' }, { status: 400 });
+  }
```

Comment:
Please use tiny invariant. It's made for that, tableId cannot be undefined or empty because it's part of the URL